solfege: add gdk-pixbuf to nativeBuildInputs

jariji opened the pull request:
###### Description of changes

Add gdk-pixbuf to `nativeBuildInputs` because otherwise it fails at startup with an exception. Likewise an error occurs at startup using `buildInputs` instead, so `nativeBuildInputs` is used.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

bobby285271 commented:
Hi, you also need to add it into args (i.e. somewhere in line 1-6) to make CI happy. But since this [builds fine on hydra](https://hydra.nixos.org/job/nixpkgs/trunk/solfege.x86_64-linux), it will be nice to provide more information for us, e.g.:

- What problem you are experiencing / why this is needed
- Why `nativeBuildInputs` and not `buildInputs`

in the "Description of changes" of your PR template and your commit message

bjornfor commented:
@jariji: I see you've fixed the code (good!), but could you also do what @bobby285271 mentioned?

> [...] to provide more information for us, e.g.:
>
> * What problem you are experiencing / why this is needed
> * [...]
>
> in the "Description of changes" of your PR template and your commit message

I tested solfege just now (on master and release-22.11), just tried detecting a few intervals, and it seemed to work fine, so I wonder what gdk-pixbuf is needed for.

bjornfor reviewed (changes requested):
Please squash commits, and add a line in the one commit message why this is needed.

bjornfor commented on pkgs/misc/solfege/default.nix:
```diff
@@ -25,11 +25,10 @@ buildPythonApplication rec {
     aclocal
     autoconf
   '';
-
```
Unrelated (and unwanted?) whitespace change.

bjornfor commented:
@jariji: What's the startup error? It works fine for me (on NixOS), both on master branch and release-22.11.

jariji commented:
```
Traceback (most recent call last):
  File "/nix/store/vaiag1zhpi2xagjmlqx4aib0c63mmjpp-solfege-3.23.4/share/solfege/solfege/startup.py", line 128, in start_gui
    solfege.win = w = MainWin(options, datadir)
  File "/nix/store/vaiag1zhpi2xagjmlqx4aib0c63mmjpp-solfege-3.23.4/share/solfege/solfege/mainwin.py", line 126, in __init__
    stock.SolfegeIconFactory(self, datadir)
  File "/nix/store/vaiag1zhpi2xagjmlqx4aib0c63mmjpp-solfege-3.23.4/share/solfege/solfege/stock.py", line 79, in __init__
    self.add_icons(d)
  File "/nix/store/vaiag1zhpi2xagjmlqx4aib0c63mmjpp-solfege-3.23.4/share/solfege/solfege/stock.py", line 35, in add_icons
    iconset = Gtk.IconSet(GdkPixbuf.Pixbuf.new_from_file(os.path.join(self.datadir, filename)))
gi.repository.GLib.GError: gdk-pixbuf-error-quark: Couldn’t recognize the image file format for file “/nix/store/vaiag1zhpi2xagjmlqx4aib0c63mmjpp-solfege-3.23.4/share/solfege/graphics/solfege.svg”
```

Commit:
solfege: Add gdk-pixbuf dependency.

An error occurs at startup without this change.
It is in nativeBuildInputs instead of buildInputs because
the fix does not work in buildInputs.

bjornfor commented:
Ah, that issue might be hidden for me because I use NixOS / GNOME. I tried to reproduce with `env -i ./result/bin/solfege`, but that failed with `RuntimeError: Gtk couldn't be initialized. Use Gtk.init_check() if you want to handle this case.`
